packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -415,23 +443,26 @@ class HttpRouteSpec extends RouteSpec {
   }
 
   public bind(scope: Construct): RouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
-    }
+    const pathMatchConfig = this.match?.path?.bind(scope);
 
+    // Set prefix path match to '/' if none of path matches are defined.
+    const prefixPathMatch = pathMatchConfig ? pathMatchConfig.prefixPathMatch : '/';
     const headers = this.match?.headers;
-    validateMatchArrayLength(headers);
+    const queryParameters = this.match?.queryParameters;
+
+    validateHttpMatchArrayLength(headers, queryParameters);
 
     const httpConfig: CfnRoute.HttpRouteProperty = {
       action: {
         weightedTargets: renderWeightedTargets(this.weightedTargets),
       },
       match: {
-        prefix: prefixPath,
+        prefix: prefixPathMatch,
+        path: pathMatchConfig?.wholePathMatch,
```

Comment:
There's one subtlety with the way you've written this method.

If a customer provides a `match` without a `path`, but with a different property, like `method`:

```ts
match: {
  method: HttRouteMethod.GET,
},
...
```

You will render a `prefix` of `'/'` in the template, even though there's no need for it (the service would set that default itself).

Not sure if this is something you want to fix - I'm not intending to block the PR on it. But if you wanted to, you could do it pretty easily with something like this:

```
    const pathMatchConfig = (allMatchPropertiesAreUndefined(this.match)
      ? HttRoutePathMatch.startsWith('/')
      : this.match?.path)?.bind(scope);
```

In other words - only setting the `path` property if no other `match` property was set.

Up to you!

Reply:
I think we can keep it this way. 

If we would like be more explicit on assigning `/` to `path` only if none of the match properties are defined, it could be more straightforward by just requiring to specify one of the match property instead of using the default.